### ISSUE TITLE
after commit --amend, push --force "The application is not running.. "

### ISSUE THREAD

**Issue by d-adamkiewicz** [nodester/nodester#72]
Hi

I'm used to use 
git commit --amend 
in order to 'commit' small changes in code, add descriptions
Next I type
git push --force nodester master
When I access the app url (http://test1410.nodester.com) I get the following:

"The application is not running..
ECONNREFUSED, Connection refused"

**Comment by davglass** [comment]
When we pull into the chroot, we do a standard pull. Which means your --force will break.

IMHO, you should never use --force and --amend is not the right thing to do in this case, you lose valuable history doing this.

If you wish to continue doing this, you need to force a reset on the git repo each time to ensure the clone is successful. `nodester app gitreset`. 

Note, doing this will cause you to lose all npm installed modules that you will have to re-install.

**Comment by davglass** [comment]
Closed due to no activity.

**Comment by JasonWoof** [comment]
I wish I'd caught this discussion as it was getting started, but I didn't know about nodester then :)

I've used a different approach to make it easy for me to "git push" to my webserver. Below I describe my main usecase, how I did it, and how it saves disk space.

I wish I could push a commit which is not a child of the last commit that was pushed. One of the things I like about tracking my website development in git is that if a client decides that the latest version of the site is undesirable (ugly, buggy, etc) I can quick add a branch, then roll master back to the last good revision, push that, then go back to my branch to fix it, and later I can merge this into master and publish it again.

I now publish my websites to my server with "git push". At first I had the .git folder in the webroot, and had a post-commit hook that did "git reset --hard master" to get the files updated. But then someone in freenode#git told me that I should instead be using a bare git repo on my server (which need not be in the webroot), and using "git checkout" to update the files in the webroot.

This new approach has worked very well for me. I've got a post-update hook in the bare repos (in my home directory on my webserver) that do basically this:

export GIT_DIR="$HOME/git-www/foo.git"
export GIT_WORK_TREE="/var/www/foo.com"
git checkout -f

I save quite a bit of disk space with this because there are a few libraries that many of my sites use (often different versions of them.) I have just one bare git repo for each library, and the post-update scripts for my sites do a "git checkout -f submodules-requested-rev" to get the correct version of the lib installed into that site's webroot.

I suspect this would save disk space for you too, because you don't need a copy of .git in the chroot

I hope this helps you improve your awesome website! I'm sorry that I don't have the time/energy to add this feature myself.

Take care,   - Jason
